src/jbi/router.py

```diff
@@ -1,34 +1,79 @@
 """
 Router dedicated to Jira Bugzilla Integration APIs
 """
+import importlib
 import logging
-from typing import Optional
+from types import ModuleType
+from typing import Dict, List, Optional
 
-from fastapi import APIRouter, Depends, Request
-from fastapi.responses import HTMLResponse
+from fastapi import APIRouter, Body, Depends, Request
+from fastapi.responses import HTMLResponse, JSONResponse
 from fastapi.templating import Jinja2Templates
 
 from src.app import environment
 from src.jbi import configuration
+from src.jbi.bugzilla_objects import BugzillaBug, BugzillaWebhookRequest
+from src.jbi.model import ActionError, IgnoreInvalidRequestError
+from src.jbi.service import get_bugzilla
 
 templates = Jinja2Templates(directory="src/templates")
 
 api_router = APIRouter(tags=["JBI"])
 
-jbi_logger = logging.getLogger("src.jbi")
+jbi_logger = logging.getLogger("src.jbi.router")
 
 
-def execute_request(request, settings):
+def extract_current_action(  # pylint: disable=inconsistent-return-statements
+    bug_obj: BugzillaBug, action_map, settings
+):
+    """Find first matching action from bug's whiteboard list"""
+    potential_configuration_tags: List[
+        str
+    ] = bug_obj.get_potential_whiteboard_config_list()
+
+    for tag in potential_configuration_tags:
+        if tag.lower() in action_map:
+            return action_map.get(tag.lower())
```

Comment:
Actually, this would be a use-case for `:=` :) 

```suggestion
        if action := action_map.get(tag.lower()):
            return action
```